### PR TITLE
Add support for broadcast ops in MIXR

### PR DESCRIPTION
**Pull request by jungpark-mlir**
contains
- migraphx.broadcast
- migraphx.multibroadcast
- fix broadcast folding with nchw layout

### CODE REVIEW

File: mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp [hunk 1]
```diff
@@ -135,9 +135,131 @@ class ConvConverter final
...
+                  ConversionPatternRewriter &rewriter) const final {
+    auto operands = adaptor.getOperands();
+    Location loc = op->getLoc();
+    auto input_t = operands[0];
+    auto axis = op->getAttr("axis").cast<IntegerAttr>().getInt();
+
+    for (auto &use : op->getResult(0).getUses()) {
+      auto expandedOp = use.getOwner();
+      // isa binary operation,
+      if (isa<migraphx::AddOp>(expandedOp) ||
```
Comment (sjw36) [inline mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp]: Shouldn't need to restrict to these ops. Can we just replace the migraphx.broadcast with the tosa.reshape?

Comment (jungpark-mlir) [reply mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp]: broadcast in tosa is restrictive. Just checked and it only broadcasts on the second input of the binary operations. Not clearly defined as a rule but only defined per each operator.
I'll change it to reflect that rule.

File: mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp [hunk 2]
```diff
@@ -135,9 +147,134 @@ class ConvConverter final
...
+
+    // get shape of the use
+    auto outShape = op->getResultTypes()[0].cast<ShapedType>().getShape();
+    auto outElemType =
+        op->getResultTypes()[0].cast<ShapedType>().getElementType();
+    uint32_t outRank = outShape.size();
+    auto inShape = input_t.getType().cast<ShapedType>().getShape();
+
+    Value newOperand = input_t;
+    if (outRank != inShape.size()) {
```
Comment (sjw36) [inline mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp:171]: Looks good.

**Review by sjw36 — APPROVED** [review]
LGTM

**Pull request merged**